i6_models/parts/conformer/frontend.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+__all__ = ["ConformerFrontendV1", "ConformerFrontendV1Config"]
+from typing import Optional
+from dataclasses import dataclass
+import torch
+from torch import nn
+
+from i6_models.config import ModelConfiguration
+
+
+@dataclass
+class ConformerFrontendV1Config(ModelConfiguration):
+    """
+    :param feature_dim: Feature dimension of the input data
+    :param hidden_dim: Hidden dimension used in the model internally
+    :param dropout: Dropout value after linear transformation
+    :param conv_stride: Stride of down-sampling cov
+    :param conv_kernel: Kernel size of down-sampling conv
+    :param conv_padding: Padding factor of down-sampling conv
+    :param spec_aug_cfg: TODO
+    """
+
+    feature_dim: int
+    hidden_dim: int
+    dropout: float
+    conv_stride: int
+    conv_kernel: int
+    conv_padding: int
+
+    # nested configurations
+    spec_aug_cfg: Optional[ModelConfiguration]  # TODO
+
+
+class ConformerFrontendV1(nn.Module):
+    """
+    Frontend part of the standard conformer doing down-sampling
+    """
+
+    def __init__(self, cfg: ConformerFrontendV1Config):
+        """
+        :param cfg: conformer frontend configuration
+        """
+        super().__init__()
+        if cfg.spec_aug_cfg:
+            # TODO: This still needs to be implemented
+            raise NotImplementedError
+        else:
+            self.spec_aug = None
+        self.subsampling = nn.Conv1d(
+            in_channels=cfg.feature_dim,
+            out_channels=cfg.feature_dim,
```

Comment:
Did you check in other implementations if this is always done like this? The paper does not tell anything about it, but we might check other implementations for reference.

Reply:
I am not 100% sure here. It seems like most frameworks handle the part before the first Conformer encoder independently and don't see it as part of the Conformer (e.g. torchaudio does it like [this](https://pytorch.org/audio/main/_modules/torchaudio/models/conformer.html#Conformer)) In one of the meetings we decided to have this as part of the conformer. we can change this and only have the "stack" as a model here and leave the sub-sampling to the user, but I would not make this change in the PR without other opinions since after all we agreed on a variant in the first place. 

From what I found we have e.g: 
https://espnet.github.io/espnet/_modules/espnet2/asr_transducer/encoder/blocks/conv_input.html#ConvInput
https://speechbrain.readthedocs.io/en/latest/_modules/speechbrain/lobes/models/ESPnetVGG.html#ESPnetVGG

In PaddleSpeech they have it as a BaseEncoder class from which the Conformer then inherits which defines the subsampling in different variations.
https://github.com/PaddlePaddle/PaddleSpeech/blob/5f53e902e1c85a7ec6c1645d61a26701d171428a/paddlespeech/s2t/modules/encoder.py#L59

So as a conclusion I think this all suggests that subsampling is not seen as part of the Conformer. So how we handle this here? Any opinions?

Reply:
I know the decision was to stay close to the paper. However, just having one layer here differs from all other implementations (the ones you linked and our own ones) and is most likely suboptimal. The question is whether we want to ignore this for now to have a working version as soon as possible and everything else will be "v2" or if we follow another implementation or if we should make it flexible enough to allow overwriting this.